packages/frontend-v2/pages/login.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  Button,
+  FormControl,
+  FormErrorMessage,
+  Input,
+  InputGroup,
+  Text,
+} from "@chakra-ui/react";
+import { API } from "@graduate/api-client";
+import { NextPage } from "next";
+import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { AxiosError } from "axios";
+import { logger, routeOnboarding } from "../utils";
+import { LoginStudentDto } from "@graduate/common";
+const Login: NextPage = () => {
+  const [apiError, setApiError] = useState("");
+  const [renderSpinner, setRenderSpinner] = useState(true);
+  const router = useRouter();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<LoginStudentDto>({
+    mode: "onTouched",
+    shouldFocusError: true,
+  });
+
+  const loginWithCookie = async () => {
+    setRenderSpinner(true);
+    try {
+      const student = await API.student.getMe();
+      if (student) {
+        if (student.isOnboarded) {
+          // Redirect to home
+          router.push("/home");
+        } else {
+          // Redirect to onboarding
+          router.push("/onboarding");
+        }
+      }
+    } catch (err) {
+      const error = err as AxiosError;
+      logger.error(error);
+      setRenderSpinner(false);
+    }
+  };
+
+  useEffect(() => {
+    loginWithCookie();
+  }, []);
```

Comment:
good use case for a custom hook "useRedirectIfLoggedIn()" that you can just call at the top of this component, it'll just clean up the component and can probably be reused in sign up too

Reply:
Hmm, I can make this a custom hook but I dont think it would be smart to do this for Signup from a UX perspective. If a user logs out and wants to create a new user but still has the JWT in cookies, then we wouldn't want to redirect them to their old page if they're trying to sign up as a new user. What are your thoughts?

Reply:
sounds good!